tree/ntuple/v7/doc/specifications.md

```diff
@@ -1,159 +1,636 @@
-# RNTuple Reference Specifications (DRAFT, WIP)
+# RNTuple Reference Specifications (WIP)
 
-## Terminology
+**Note:** This is work in progress. The RNTuple specification is not yet finalized.
 
-### Column
-A set of data values, all of a single fundamental type, which is a storage-backed array.
+This document describes version 1 of the RNTuple binary format.
 
-### Page 
-Partitioned columns, typically a few tens of kilobytes in size. The smallest addresable unit.
+The RNTuple binary format describes the serialized, on-disk representation of an RNTuple data set.
+The data on disk is organized in **pages** (typically 10-100kB in size)
+and several **envelopes** that contain information about the data such as header and footer.
+The RNTuple format specifies the binary layout of the pages and the envelopes.
 
-### Cluster
-A set of pages that contain all the data of a certain entry range. They are typically a few tens of megabytes
-in size and a natural unit of processing for a single thread or task.
+Pages and envelopes are meant to be embedded in a data container
+such as a ROOT file or a set of objects in an object store.
+Envelopes can reference other envelopes and pages by means of a **locator** or an **envelope link**;
+for a file embedding, the locator consists of an offset and a size.
+The RNTuple format does _not_ establish a specific order of pages and envelopes.
 
-### Collection
-Vector that allows for variable-length storage of elements (which can be arbitrarily complex, e.g., collection of record or collection of collection) within a row.
+Every emedding must define an **anchor** that contains the envelope links (location, compressed and uncompressed size)
```

Comment:
```suggestion
Every embedding must define an **anchor** that contains the envelope links (location, compressed and uncompressed size)
```

Reply:
A couple of things for a subsequent PR:
- I think this overview could be extended to provide more information on how all the pieces fit together.
- IMO, the ASCII art below for several structures should also show conditional / variable size members.